pipeline_dp/dp_engine.py

```diff
@@ -62,7 +62,9 @@ def aggregate(self, col, params: pipeline_dp.AggregateParams,
         _check_aggregate_params(col, params, data_extractors)
 
         with self._budget_accountant.scope(weight=params.budget_weight):
-            return self._aggregate(col, params, data_extractors)
+            result = self._aggregate(col, params, data_extractors)
+            self._backend.annotate()
```

Comment:
`annotate()` takes col, returns col and takes additional args (budget_scope for now)

```
with self._budget_...scope(...) as scope:
   return self._backend.annotate(col, scope)
```